### DISCUSSION TITLE
GitExtensions Evaluation Questions

### DISCUSSION THREAD

**Discussion by Jan-Thes** [gitextensions/gitextensions#9090] [General]
Hello,
im currently evaluating GitExtensions for use in my company project, and it has a bunch of strengths when compared to EGit, SourceTree, TortoiseGit, Git GUI and Fork. I have on the other hand noticed some problems and i wanted to ask if i did things wrong or if those are simply open issues.

- Unlike any other client i have evaluated, GitExtensions seems to ignore global git settings. This is a blocker for us because we need to keep in sync the Git settings of all clients (we will always use EGit in addition to a standalone client) of all our developers. Examples for ignored global setttings:
  - `pull.rebase`
  - `user.name`, `user.email` (it asked for these on installation even though they were already set)
  - `branch.develop.mergeoptions=--no-ff` and similar things are not respected when doing a merge (this setting means that when merging any branch into a branch called `develop`, the -no-ff flag needs to be on by default)
  - we need the `-x` flag for cherry-pick to be on for everybody somehow  
  
   So it seems like we have not choice but to manually configure GitExtensions and then use a script to copy `GitExtensions.settings` to every developer?
  
- It doesn't show tags by default? I have found how to display them but i was seriously confused on why those are not on by default.

- It does not seem to allow to make annotated tags the default. Since annotated tags are the default in SourceTree (which is very popular all over our company), not having this means we would end up with a mix of leightweight and annotated tags.

- We often enough have the need to search for commits by path of changed files. It seemt really weird to me that this is possible only when going to View- > Set Advanced Filters -> File Filter, when pretty much every other option is directly available in the filter text input field in the icon bar. Not a deal breaker at all, just wondering if there is a reason for this.

I think that GitExtensions could be a very popular Git client for average software companies, especially since it has File Filter, which most other clients don't have properly, so maybe this is some valuable feedback for you about the kind of things that still block this client from being selected, at least for our use cases, where most developers really are not familiar enough with Git to configure all the settings by themselves.

**Comment by gerhardol** [comment]
Global settings: Set home
There is a GUI to set email, pull.rebase too. The UI is not updated from the defaults to mergeoptions etc though.
![image](https://user-images.githubusercontent.com/6248932/115307553-abbf2180-a169-11eb-89d9-948e2872511b.png)

Show tags:
The code has true, the default config false. Could be changed.
Probably no decision to why it is like that, GE is driven by user changes.

Tags:
Personally, I would not vote no to changing default to annotated tags, that is how my usage is anyway. There may be users that use lightweight tags.

File filters:
File searches etc has been discussed, the conclusion is that the filters should be changed. File filters was discussed there.

GE is community driven, features that someone want a lot is implemented before those that many want a little.
PRs are accepted.

**Reply by Jan-Thes** [reply]
Ok thanks for the info!

Global Setting:
I don't understand how GitExtensions interacts with the home directory. It does not seem to read the .gitconfig file anyway, so why do you need to configure a home directory? Does it write to the .gitconfig file but not read?

Tags:
I don't want annotated tags to be the default always, i was just wondering if there is a way to configure the default.
